index.js

```diff
@@ -1,19 +1,35 @@
 'use strict';
 
 var isObject = require('isobject');
+var isArray = require('isarray')
 var forOwn = require('for-own');
 var omit = require('omit-keys');
 
-module.exports = function omitDeep(obj, props) {
-  var o = {};
 
-  forOwn(obj, function (value, key) {
-    if (isObject(value)) {
-      o[key] = omitDeep(value, props);
-    } else {
-      o[key] = value;
+module.exports = function omitDeep(input, props) {
+  function omitDeepOnOwnProps(obj) {
+    if (!isArray(obj) && !isObject(obj)) {
+      return obj
```

Comment:
Whoops! Sorry about that. That's the convention we have at work, and I wasn't paying attention :)